server/routes/choropleth.py

```diff
@@ -35,85 +35,109 @@
+        perCapita -> Whether to return the per-capita value, as a string of a 
+                boolean.
+        geoId -> The currently viewed geography to render.
+        level -> The subgeographic level to pull and display information for.
+        mdom -> The measurement denominator to use if perCapita is true. 
+                Defaults to Count_Person.
+
 
     API Returns:
-        df -> dictionary of statistical variable values by country.
-            Additionally includes some plotting information.
+        df -> Returns a dictionary with two entries
+            geoJson -> geoJson format that includes statistical variables info,
+                    geoId, and name for all subregions.
+            _PLOTTING_INFO -> Includes domain range for use in D3 and the
+                    color palette to render with.
     """
     # Get required request parameters.
-    stat_var_requested = request.args.get("statVar")
-    if stat_var_requested is None:
+    requested_stat_var = request.args.get("statVar")
+    if requested_stat_var is None:
         return jsonify({"error": "Must provide a 'statVar' field!"}, 400)
-    place_requested = request.args.get("place")
-    if place_requested is None:
-        return jsonify({"error": "Must provide a 'place' field!"}, 400)
+    requested_geoId = request.args.get("geoId")
+    if requested_geoId is None:
+        return jsonify({"error": "Must provide a 'geoId' field!"}, 400)
 
     # Get optional fields.
-    per_capita = request.args.get("perCapita", default=False) == "true"
-
-    # Get the type of the requested place. Accept first valid type.
-    geolevel_below = dc.get_property_values([place_requested],
-                                            "typeOf")[place_requested]
-    for level in geolevel_below:
-        if level in LEVEL_MAP:
-            geolevel_below = level
+    per_capita = request.args.get("perCapita", default=False) in (
+        [True, "true", "True", 1, "t"])
+    display_level = request.args.get("level")
+    measurement_denominator = request.args.get("mdom", default="Count_Person")
+
+    # If no level is requested then default to one level below current.
+    if display_level is None:
+        requested_geoId_type = dc.get_property_values([requested_geoId],
+                                                "typeOf")[requested_geoId]
+        for level in requested_geoId_type:
+            if level in LEVEL_MAP:
+                display_level = LEVEL_MAP[level]
+                break
 
     # Get list of all contained places.
     geos_contained_in_place = dc.get_places_in(
-            [place_requested], LEVEL_MAP[geolevel_below])[place_requested]
+            [requested_geoId], display_level)[requested_geoId]
 
     # Download statistical variable, names, and geojson for subgeos.
     stat_var_by_geo = dc.get_stats(geos_contained_in_place,
-                                   stat_var_requested)
+                                   requested_stat_var)
```

Comment:
Add after line 85:
\# TODO(iancostello): Handle failing function calls.  Also, handle the case where only a fraction of values are returned.

Reply:
Added.